src/commands/installLanguageServer.ts

```diff
@@ -0,0 +1,126 @@
+import * as vscode from "vscode";
+import { Command, CommandType } from "./command";
+import * as fs from 'fs';
+import * as Path from 'path';
+import * as tar from 'tar';
+import * as Octokit from '@octokit/rest';
+import { HttpClient } from 'typed-rest-client/HttpClient';
+import { QuickPickItem } from "vscode";
+import { getConfiguration } from "../configuration";
+import { ExperimentalLanguageClient } from "../languageclient";
+
+export class InstallLanguageServerCommand extends Command {
+  public static readonly CommandName = "installLanguageServer";
+
+  constructor(ctx: vscode.ExtensionContext) {
+    super(InstallLanguageServerCommand.CommandName, ctx, CommandType.PALETTE);
+  }
+
+  protected async perform(releaseId: string = null): Promise<void> {
+    await ExperimentalLanguageClient.stopIfRunning();
+
+    const serverLocation = getConfiguration().languageServer.pathToBinary || Path.join(this.ctx.extensionPath, "lspbin");
+
+    // Create dir for lsp binary if it doesn't exist
+    if (!fs.existsSync(serverLocation)) {
+      fs.mkdirSync(serverLocation);
+    }
+    const downloadedZipFile = Path.join(serverLocation, 'terraform-lsp.tar.gz');
+    // Download language server
+    const octokit = new Octokit();
```

Comment:
the integs are failing because of rate-limiting on macos (probably because the linux started earlier): we probably want to add some simple retry and back-off logic here to ensure that this works consistently

Reply:
So looking at the API restrictions they're pretty broad. 

```
For unauthenticated requests, the rate limit allows for up to 60 requests per hour. Unauthenticated requests are associated with the originating IP address, and not the user making requests.
```
So it feels like backing off for an hour might not be a good option!

One option is to require users to login to github. I don't think this is a good UX so ruled that out. 

The second option I've gone with is to limit the number of calls I'm making and be much more efficient. I've

1. Halved the number of calls calls when doing at `Install`.
2. Made `Update` only check once a day (think this will have a big impact when running integration tests as this may have been firing for each test). 

Hopefully this get this sorted. Will test today and see if the rate limit issue re-occurs with these changes.